api/src/tool/console/interactive.py

```diff
@@ -0,0 +1,128 @@
+#
+# Interactive console for using database and services.
+#
+# Connects to the database then drops into a Python interactive prompt.
+#
+# Use via the `-i` flag to python, for example:
+#   poetry run python3 -i -m src.tool.console.interactive
+#
+
+import logging
+from typing import Union
+
+import rich
+import rich.panel
+import rich.pretty
+import sqlalchemy
+import src.db
+import src.db.models
+import src.logging
+import src.util
+import tests.src.db.models.factories
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+from src.adapters.db.clients.postgres_config import get_db_config
+
+INTRO = """
+Simpler Grants Gov Python console
+
+Useful things:
+
+│ db = database session (Not yet working)
+│ f = DB factories module
+│ u = utilities module
+│ r = function to reload REPL
+
+Tips:
+★ Tab-completion is available
+★ History is available (use ↑↓ keys)
+★ Use Ctrl+D to exit
+"""
+
+
+def interactive_console() -> dict:
+    """Set up variables and print a introduction message for the interactive console."""
+
+    db = connect_to_database()
+
+    print(INTRO.format(**locals()))
+
+    variables = dict()
+    variables.update(vars(src.db.models.opportunity_models))
+
+    # This goes after the imports of entire modules, so the console reserved
+    # names (db, fineos, etc) take precedence. This might break some modules
+    # that expect something different under those names.
+    # variables.update(locals())
+
+    # DB Factories
+    factories_module = tests.src.db.models.factories
+
+    if isinstance(db, sqlalchemy.orm.scoped_session):
+        factories_module.db_session = db
+
+    variables["f"] = tests.src.db.models.factories
```

Comment:
FYI - we might want to adjust where this utility lives - I made it so the factories are only a test component with the dependencies as dev only. We shouldn't be importing them into the `src` path ideally. This code won't ever run non-locally, but if you tried to do so, it would complain about missing packages.

I'm not sure if the `tests` folder is quite the right place, but I did put a utility script in `tests/lib/seed_local_db.py` - might be worth coming up with a happier approach.

Reply:
I'm open but would propose splitting that piece out to a separate PR.

I do think it would make sense to close the loop on the intercepting session errors - so that we don't have to manually do a rollback when it gets to an error state

Reply:
I looked into the rollback automation, not finding a ton of approaches. Tried rolling back automatically when an error happens and that just cascades into other errors. In short, SQLAlchemy changed sessions to generally be made for handling a single set of transactions (great for an API, less so for an interactive console). 

Without thinking through it much, the right approach probably involves lambdas/sessionmakers to automatically make new sessions when old ones error, but that's likely something that will take a while to figure out

Reply:
Want to just approve and merge then?  We've got db queries and ability to play with factories - that's a win I think.

Reply:
Yeah, we can put that somewhere in the backlog, just adjust the repr bit and we can go from there